### DISCUSSION TITLE
Can Maui support DynamicIsland and AppWidget on IOS Projects ?

### DISCUSSION THREAD

**Discussion by omarhagoche** [dotnet/maui#12506] [Q&A] [answered]
Hello, I Searched alot but there's nothing on internet about doing AppWidget Using .NET Maui or Xamarin.Forms Directly with out using xcode at less.

the important thing is dynamic island on ios

Is there Any one knows how ingrate DynamicIsland in an .NET Maui or Xamarin.Forms App ?

Sorry for format of my writes, I'm not native english speaker and this's my first discussion on github

Thank You

**Comment by bobbydharrell** [comment]
It would be nice for Microsoft to have a more integrated solution in MAUI or even Xamarin.Forms for both iOS and Android widget support, especially with Apple's new interactive widgets on iOS 17. Dynamic Island, watch support etc. All from within Xamarin/Maui

**Reply by Xyncgas** [reply]
Do you want drawing native IOS layout in a cross platform project

Because, if you just want cross platform there's PWA. If you want MAUI to be really customizable for every platform with a generic 2000ft abstraction of 1 line button I would just go native from the beginning, not that they are the same because you can still write one liner button for every platform, and by the way cost is not a problem for me because I can maintain multiple project at once instead of _**a more or less unified**_ code base for UI

Meanwhile, to be constructive other than saying what I want or don't want, I say one way to implement a cross platform UI is to really be cross platform, eliminate all the native UI support, implement ways to write UI and the kinds of UI possible only on this platform that runs everywhere, for example we can emulate stuffs we consider cool or useful on other platform to bring it to this unified platform, we can start implementing api like how game dev add new button or feature to their games, to say for example today we let people draw this UI that display a couple string in a list in a brand new way - - - [Redacted] (you know what idc, microsoft can hire me or they can figure it out themselves)

**Reply by Xyncgas** [reply]
All the OS venders are to be blamed too, because they are not using a single UI, they can just not let you do something because your **_cross platform UI_** is living in their OS

Maybe industry should unite on the UI part of their OS to get a consensus of how are we going to let people interact with our devices, make it into a standard and everyone can use it to write program that intercept gestures or read sms or display animation that's hardware accelerated by just providing the next position and dimension information of this UI element

This way we get truly cross platform UI

They can do this by implementing a virtual machine

Then adaptors can bind to this virtual machine in any language by giving the correct number with bits and the correct values of bits as instructions and datas

UI is incredibly functional-oriented, we had proof from frameworks that are FO which're drawing UI today, so we simply can make the assumption that functions and lambda calculus and virtual machines go hand in hand so it should be easy to do in fact I've already done it in my framework that can serialize out the layout specification I drew and interpret it in another language say C# to Python

**Reply by bobbydharrell** [reply, marked as answer]
Eh, can prolly just do an extension in swift like we currently do th3 widgets and watches. Or a renderer. Microsoft could make it easy with platform specific apis,  but in the end it doesn't matter.

**Comment by omarhagoche** [comment]
thank you every one, after 1year, finally I create my first Demo ios widget 

https://x.com/Omar_Hagoche/status/1802097874404049285

I'm looking around for how to make thins easy to implement

thnx, again.

**Reply by gsgou** [reply]
greta job @omarhagoche. Any chance you share yr sample on github?

**Reply by KaustavCodes** [reply]
This is awesome. Is this code available on Github somewhere.
